miden/src/cli/data.rs

```diff
@@ -480,3 +508,90 @@ impl Libraries {
         Ok(Self { libraries })
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::InputFile;
+
+    #[test]
+    fn test_merkle_data_parsing() {
+        let program_with_pmt = "
+        {
+            \"operand_stack\": [\"1\"],
+            \"merkle_store\": [
+                {
+                    \"partial_merkle_tree\": [
+                        [
+                            [
+                                2,
+                                0
+                            ],
+                            \"1400000000000000000000000000000000000000000000000000000000000000\"
+                        ],
+                        [
+                            [
+                                2,
+                                1
+                            ],
+                            \"1500000000000000000000000000000000000000000000000000000000000000\"
+                        ],
+                        [
+                            [
+                                1,
+                                1
+                            ],
+                            \"0b00000000000000000000000000000000000000000000000000000000000000\"
+                        ]
+                    ]
+                }
+            ]
+        }";
+        let inputs: InputFile = serde_json::from_str(&program_with_pmt).unwrap();
+        let merkle_store = inputs.parse_merkle_store().unwrap();
+        assert!(merkle_store.is_some());
+
+        let program_with_smt = "
+        {
+            \"operand_stack\": [\"1\"],
+            \"merkle_store\": [
+              {
+                \"sparse_merkle_tree\": [
+                  [
+                    0,
+                    \"1400000000000000000000000000000000000000000000000000000000000000\"
+                  ],
+                  [
+                    1,
+                    \"1500000000000000000000000000000000000000000000000000000000000000\"
+                  ],
+                  [
+                    3,
+                    \"1700000000000000000000000000000000000000000000000000000000000000\"
+                  ]
+                ]
+              }
+            ]
+          }";
+        let inputs: InputFile = serde_json::from_str(&program_with_smt).unwrap();
+        let merkle_store = inputs.parse_merkle_store().unwrap();
+        assert!(merkle_store.is_some());
+
+        let program_with_merkle_tree = "
+        {
+            \"operand_stack\": [\"1\"],
+            \"merkle_store\": [
+                {
+                    \"merkle_tree\": [
+                        \"1400000000000000000000000000000000000000000000000000000000000000\",
+                        \"1500000000000000000000000000000000000000000000000000000000000000\",
+                        \"1600000000000000000000000000000000000000000000000000000000000000\",
+                        \"1700000000000000000000000000000000000000000000000000000000000000\"
+                    ]
```

Comment:
Now looking at this, I think we should use "0x" prefix fox hex values to make it consistent with how we use the `push.0x...` instruction.